Vagrantfile

```diff
@@ -1,11 +1,57 @@
 Vagrant.configure("2") do |config|
 
-  config.vm.define "controlnode" do |web|
+  # Install required plugins
+  required_plugins = %w( vagrant-hostmanager )
+  plugin_installed = false
+  required_plugins.each do |plugin|
+    unless Vagrant.has_plugin?(plugin)
+      system "vagrant plugin install #{plugin}"
+      plugin_installed = true
+    end
+  end
+
+  # If new plugins installed, restart Vagrant process
+  if plugin_installed === true
+    exec "vagrant #{ARGV.join' '}"
+  end
+  config.vm.box_check_update = true
+  config.vm.graceful_halt_timeout=15
+  config.vbguest.auto_update = false
```

Comment:
Thanks @bbaassssiiee.

I'm seeing (On Linux) this error when running `vagrant up`:
```
An error occurred while executing the action on the 'controlnode'
machine. Please handle this error then try again:

There are errors in the configuration of this machine. Please fix
the following errors and try again:

Vagrant:
* Unknown configuration section 'vbguest'.
```